ecommerce/EarningController.php

```diff
@@ -0,0 +1,300 @@
+<?php
+/**
+ * Manage earnings
+ *
+ * @package Tutor\Ecommerce
+ * @author Themeum <support@themeum.com>
+ * @link https://themeum.com
+ * @since 3.0.0
+ */
+
+namespace Tutor\Ecommerce;
+
+use Tutor\Helpers\QueryHelper;
+use Tutor\Models\OrderModel;
+
+/**
+ * Manage earnings
+ */
+class EarningController {
+
+	/**
+	 * Error message for the invalid earning data
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var string
+	 */
+	const INVALID_DATA_MSG = 'Invalid earning data';
+
+	/**
+	 * EarningController instance
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var EarningController object
+	 */
+	public static $instance = null;
+
+	/**
+	 * Earning table name
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var string
+	 */
+	private $earning_table;
+
+	/**
+	 * Order id
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var int
+	 */
+	private $order_id;
+
+	/**
+	 * Keep earning data here
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	public $earning_data = array();
+
+	/**
+	 * Private construct to prevent direct object creation
+	 *
+	 * @since 3.0.0
+	 */
+	private function __construct() {
+
+	}
+
+	/**
+	 * Instance of this class
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return EarningController object
+	 */
+	public static function instance() {
```

Comment:
Instead of making individual class as singleton, we can use reusable `Singleton` class to make this class as singleton. PR https://github.com/themeum/tutor/pull/924

```php
class EarningController extends Singleton {

  protected function __construct() {
     global $wpdb;
     self::$instance->earning_table = $wpdb->prefix . 'tutor_earnings';
  }

```